### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by radumereuta**
Based on: https://github.com/marketplace/actions/publish-vs-code-extension
This seems to be the recommended way of automatically publishing the extensions: https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#github-action

I removed the automatic trigger. I want to trigger it manually for now.

Fixes: #36

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - run: cd k-vscode
+      - run: npm ci
+      - name: Publish extension to Visual Studio Marketplace
+        uses: HaaLeo/publish-vscode-extension@v1
+        id: publishVSMarketplace
```
Comment (ehildenb) [inline .github/workflows/publish.yml:16]: This should be added to the `release.yml` file instead, and run after the github release is published: https://github.com/runtimeverification/k/blob/master/.github/workflows/release.yml#L345

See how the `gh-pages` step depends on the `release` step, this should be done similarly.

**Review by Baltoli — COMMENTED** [review]
The actions themselves look fine wrt. publishing to the marketplaces, but I agree with @ehildenb that we should tag a Github release at the same time.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          packagePath: ''
+      - name: Attach vsix to release
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ${{ steps.create_vsix.outputs.vsixPath}}
+          asset_name: ${{ steps.create_vsix.outputs.vsixPath}}
+          asset_content_type: application/vsix
```
Comment (radumereuta) [inline .github/workflows/release.yml:58]: Based on: https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/master/.github/workflows/main.yml

**Review by radumereuta — COMMENTED** [review]
I've updated the workflow to to something similar to the kubernetes extension.
It's a single job. The action that builds the extension also publishes it to the marketplace.
So I don't see many benefits from having two separate steps like Everett suggested (but I may miss something).
@F-WRunTime please review as well. Keep in mind that this is my first workflow file. Please pay extra attention. It's likely to fail since the extension files are not in the root of the repo. So I may need to update some paths after this is merged.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**